projectile.el

```diff
@@ -1480,6 +1481,7 @@ With a prefix ARG invalidates the cache first."
 (defvar projectile-haskell-cabal '("*.cabal"))
 (defvar projectile-rust-cargo '("Cargo.toml"))
 (defvar projectile-r '("DESCRIPTION"))
+(defvar projectile-npm '("package.json" "node_modules"))
```

Comment:
Well I have to agree but only partially.
Doc string for this function states: 'Determine the project's type based on its structure.'
I know that 'package.json' is enough to detect project as NPM. So, no 'or', just '(defvar projectile-npm '("package.json"))' only. Thing is that 'node_modules' is hard coded and characteristic structural element for NPM projects and this is why I decided to add this directory. NPM projects without 'node_modules' are rare. 

If you think that 'package.json' is enough to detect project type as NPM then I'm not going to argue :)

Different kind of problem is that Projectile does not support multiple "project natures" as for example Eclipse does.

There is a lot of projects which have: 'bower.json', 'package.json' and 'grunt.js' or 'gulpfile.js' at the some time and it's pretty hard to find if they are NPM packages with bower support or bower packages with NPM used for dependencies or both.